devices/cpu/registers.h

```diff
@@ -39,6 +41,25 @@ typedef struct Status_reg {
   uint16_t reserved : 7;   // Reserved bits
 } __attribute__((packed)) Status_reg;
 
+// Structure describing call information //
+typedef struct CallEntry {
+  uint16_t targetPc; // Target call PC
+  uint16_t returnPc; // Return PC (one instruction after the call)
+  uint16_t sp;       // SP value at the time of the call
+} CallEntry;
+
+// Structure containing data for call tracing //
+typedef struct CallTracer {
```

Comment:
Why not std::stack<CallTracer_MaxCallDepth>?

Reply:
std::  looks like a namespace; <> looks like a template. These are C++ stuff, and the emulator seems to be written in C,.. A C++ rewrite is not my intention.